builder/dockerfile/dispatchers.go

```diff
@@ -71,6 +71,10 @@ func env(b *Builder, args []string, attributes map[string]bool, original string)
 			return errBlankCommandNames("ENV")
 		}
 
+		if runtime.GOOS == "windows" {
+			// Case insensitive environment variables on Windows
+			args[j] = strings.ToUpper(args[j])
```

Comment:
They are not case insensitive in a way that you can query all environment variables and you will see their keys in the original form. Not normalized to any specific form or uppercased. This means that although `$env:foobar` and `$env:FOOBAR` have the same value an application can specifically check if `foobar` is in the environment without matching the uppercase form.